unity-client/Assets/Scripts/MainScripts/DCL/Components/Avatar/BodyShapeController.cs

```diff
@@ -137,6 +137,22 @@ private Animation PrepareAnimation(GameObject container)
             container.transform);
 
         createdAnimation.cullingType = AnimationCullingType.BasedOnRenderers;
+
+        //NOTE(Mordi): Adds audio handler for animation events, and passes in the audioContainer for the avatar
+        AvatarAnimationEventAudioHandler animationEventAudioHandler = createdAnimation.gameObject.AddComponent<AvatarAnimationEventAudioHandler>();
+        AudioContainer audioContainer = container.transform.parent.parent.GetComponentInChildren<AudioContainer>();
+        if (audioContainer != null)
```

Comment:
I believe this is the check to know if the avatar is remote or not. From now I would move this to a method such as:

```
bool IsRemoteAvatar(GameObject container, out AudioContainer audioContainer)
{
     audioContainer = container.transform.parent.parent.GetComponentInChildren<AudioContainer>();
     return audioContainer != null;
}
```
It would encapsulate the functionality and increasing readability.

In the future we should have a better way of knowing that an Avatar is remote rather than just looking for this AudioContainer.

Reply:
I moved this into a method called "Initialize Audio Event Handlers", which should make it a bit clearer at least. It doesn't really check whether the avatar is remote - but a few lines below it checks if the audio handler for remote avatars exists in the object and then initializes it.